osgp/protocol-adapter-mqtt/osgp-protocol-simulator-mqtt/src/main/java/org/opensmartgridplatform/simulator/protocol/mqtt/Simulator.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.simulator.protocol.mqtt;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Properties;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.moquette.broker.config.MemoryConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.io.ClassPathResource;
+
+public class Simulator {
+
+    private static final Logger LOG = LoggerFactory.getLogger(Simulator.class);
+    private static final String DEFAULT_BROKER_HOST = "127.0.0.1";
+    private static final int DEFAULT_BROKER_PORT = 8883;
+
+    public static void main(final String[] args) throws Exception {
+        final Simulator app = new Simulator();
+        app.run(args);
+    }
+
+    private void run(final String[] args) throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS00112)

Reply:
Done